ciao/tool/show.go

```diff
@@ -0,0 +1,115 @@
+package tool
+
+import (
+	"bytes"
+
+	"github.com/ciao-project/ciao/ciao-controller/types"
+	"github.com/ciao-project/ciao/ciao-controller/api"
+	"github.com/ciao-project/ciao/client"
+
+	"github.com/pkg/errors"
+)
+
+/* Intended calling convention by user would be:
+ * instances := tool.Show(ClientHandle, "instance", nil)
+ * for example, to return a string of all instances using
+ * the default tfortools template {{table .}}. This can
+ * then be printed or parsed as needed. */
+func Show(c *client.Client, objName string, data CommandOpts) (bytes.Buffer, error) {
```

Comment:
I think this should be split into List/Show and each type should be their own flags. With that I think it makes sense not to try and generalise this code and instead just do it for each object type separately.